File: leetcode/buysellstocks.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int left = 0, right = 1, maks = 0;
        int n = prices.size();
        while (right < n)
        {
            if (prices[left] < prices[right])
            {
                maks = max(maks, prices[right] - prices[left]);
            }
            else
            {
                left = right;
            }
            right++;
        }
        return maks;
    }
};

File: leetcode/buysellstocks.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0, 1
        maks = 0
        while r < len(prices):
            if prices[l] < prices[r]:
                maks = max(maks, (prices[r] - prices[l]))
            else:
                l = r
            r += 1
        return maks    
            
